src/system/SystemConfig.h

```diff
@@ -771,3 +771,20 @@ struct LwIPEvent;
 #define CHIP_SYSTEM_CONFIG_USE_ZEPHYR_EVENTFD 0
 #endif
 #endif // CHIP_SYSTEM_CONFIG_USE_ZEPHYR_EVENTFD
+
+/**
+ * @def CHIP_SYSTEM_CONFIG_LWIP_PBUF_FROM_CUSTOM_POOL
+ *
+ * @brief
+ *     Enable the use of lwIP pbufs from a custom pool
+ *
+ * This config option exist because not all platforms defines LWIP_PBUF_FROM_CUSTOM_POOLS in lwip/opt.h
+ * Defaults to LWIP_PBUF_FROM_CUSTOM_POOLS if defined, otherwise 0
+ */
+#ifndef CHIP_SYSTEM_CONFIG_LWIP_PBUF_FROM_CUSTOM_POOL
+#if defined(LWIP_PBUF_FROM_CUSTOM_POOLS)
```

Comment:
For this to work, doesn't this file need to include `<lwip/opt.h>` as needed?  This seems to be depending on include order, which we do not want.

Also, if we do this, then the lwip/opt.h includes that were added for uses of LWIP_PBUF_FROM_CUSTOM_POOLS should be removed in the places that are including this header and using CHIP_SYSTEM_CONFIG_LWIP_PBUF_FROM_CUSTOM_POOL instead.

Reply:
This file already includes the `<lwip/opt.h>`, I moved newly added code just below that for easier reading. Also, addressed the other suggestions.